### PR TITLE
Closes #8956 - cinnamon-settings.py and ExtensionCore.py: now compatibles with Python 3.8

### LINKED ISSUE #8956: cinnamon-settings applets crashes
* Cinnamon version (cinnamon --version) **Cinnamon 4.2.4**
   - Please specify if you are using the daily builds PPA (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds). **No**
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) **Arch**
 * Graphics hardware *and* driver used **Vega 64 Mesa 19.2.4**
 * 32 or 64 bit **64 bit**
 * Attach /home/<username>/.xsession-errors, or /var/log/sysl
[xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/3848378/xsession-errors.txt)

og

**Issue**

cinnamon-settings applets crashes

**Steps to reproduce**
launch cinnamon-settings applets

**Expected behaviour**

**Other information**

```
> $ cinnamon-settings applets                                                                                                                                  
True
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:421: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if category.int_name is not "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:430: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if category.int_name is "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:445: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if category.int_name is "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:455: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if entry is not "_invalid_":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:538: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if cat[2].int_name is "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:576: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if cat[2].int_name is "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:602: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if cat[2].int_name is "custom":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if entry is not "":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:733: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if entry is not "":
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:779: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  ok_enabled = self.name_entry.get_text().strip() is not "" and self.command_entry.get_text().strip() is not ""
/usr/share/cinnamon/cinnamon-settings/modules/cs_keyboard.py:779: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  ok_enabled = self.name_entry.get_text().strip() is not "" and self.command_entry.get_text().strip() is not ""
Using pam module (python3-pampy)
/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py:423: DeprecationWarning: Gtk.Window.set_wmclass is deprecated
  self.window.set_wmclass(wm_class, wm_class)
Loading Applets module
Traceback (most recent call last):
  File "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", line 724, in <module>
    window = MainWindow()
  File "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", line 428, in __init__
    self.go_to_sidepage(cat, path, user_action=False)
  File "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", line 185, in go_to_sidepage
    sidePage.build()
  File "/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py", line 123, in build
    self.module.on_module_selected()
  File "/usr/share/cinnamon/cinnamon-settings/modules/cs_applets.py", line 21, in on_module_selected
    self.sidePage.load(self.window)
  File "/usr/share/cinnamon/cinnamon-settings/modules/cs_applets.py", line 46, in load
    download_applets_page = DownloadSpicesPage(self, self.collection_type, self.spices, window)
  File "/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py", line 810, in __init__
    self.build_list()
  File "/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py", line 897, in build_list
    msg_text = _("Update all") + ' (' + ngettext("%d update available", "%d updates available", updates_available) % updates_available  + ')'
NameError: name 'ngettext' is not defined
```

### PR DESCRIPTION
**Pull request by claudiux**
(Python 3.8 is used on Arch with Cinnamon 4.2.4.)
Also replaces `self.window.set_wmclass(wm_class, wm_class)` by `self.window.set_role(wm_class)` to avoid the warning message "Gtk.Window.set_wmclass is deprecated" (https://developer.gnome.org/gtk3/stable/GtkWindow.html#gtk-window-set-wmclass).

Closes #8956.

### CODE REVIEW

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Cinnamon/Cinnamon/pullRequest?prid=4511036&bid=15199947)

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Cinnamon/Cinnamon/pullRequest?prid=4511036&bid=15199947)

File: files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import os
 import re
 import json
-import cgi
+import html
```
Comment (eli-schwartz) [inline files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py]: This is already fixed on master, please rebase your PR and remove the last two commits.

File: files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 import SettingsWidgets
 
 # i18n
-gettext.install("cinnamon", "/usr/share/locale", names="ngettext")
+gettext.install("cinnamon", "/usr/share/locale", names="gettext.ngettext")
```
Comment (eli-schwartz) [inline files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py]: Both `"ngettext"` and `"gettext.ngettext"` are incorrect, they are the set sequence `{'t', 'x', 'e', 'n', 'g', '.'}` and have no syntactic meaning here. Additionally, your later on ngettext is changed to be accessed via gettext.ngettext, so it doesn't need the install() function to install a global namespace object.

File: files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py [hunk 3]
```diff
@@ -422,7 +422,7 @@ def __init__(self):
             # can consider it as a standalone app and give it its own
             # group.
             wm_class = "cinnamon-settings %s" % arg1
-            self.window.set_wmclass(wm_class, wm_class)
+            self.window.set_role(wm_class)
```
Comment (eli-schwartz) [inline files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py]: Removing a deprecation sounds useful, but why does this come with a commit message claiming it is relevant to python 3.8?

File: files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py [hunk 4]
```diff
@@ -910,7 +910,7 @@ def build_list(self, *args):
         updates_available = self.spices.get_n_updates()
         self.update_all_button.set_sensitive(updates_available)
         if updates_available > 0:
-            msg_text = _("Update all") + ' (' + ngettext("%d update available", "%d updates available", updates_available) % updates_available  + ')'
+            msg_text = _("Update all") + ' (' + gettext.ngettext("%d update available", "%d updates available", updates_available) % updates_available  + ')'
```
Comment (eli-schwartz) [inline files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py]: I'd intuitively expect any future users of the API to want to use ngettext() directly, but maybe that is just me.

File: files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py (hunk 1 above)
Comment (claudiux) [reply files/usr/share/cinnamon/cinnamon-settings/bin/ExtensionCore.py]: Done.

**Comment by clefebvre** [comment]
I'll close this for now but we should chase it and fix it all the same. The reason I close it is because neither fixes work here but more importantly they shouldn't be part of the same PR.

**Window role:**

I don't mind something more modern than `self.window.set_wmclass(wm_class, wm_class)` but `self.window.set_role(wm_class)` doesn't seem to work as well. Or if it does, then we also need to modify GWL (the grouped-window-list applet) to take the change into account.

As it is, when you launch `cinnamon-settings applets` from master you get an applets icon in GWL which is the same as in your alt-tab selector. With this code change, this is broken and GWL shows the generic icon for System Settings.

**Ngettext**

What's going on with Python 3.8 and ngettext? Where is the regression coming from? Testing here with python 3.6 in French, on master I get "9 mises a jour disponibles".. if I switch from `gettext` to `gettext.ngettext` or if I add `from gettext import ngettext`, it fails to fetch the translation and I end up with the English "9 available updates".

**Comment by eli-schwartz** [comment]
@clefebvre I rationalized this in #8964

**Pull request closed without merging**